packages/core/src/delegation/DelegationDecoder.ts

```diff
@@ -81,35 +79,50 @@ function decodePermissions(bitset: number): Permission[] {
 
 export type DelegationNodeRecord = Pick<
   DelegationNode,
-  'rootId' | 'parentId' | 'account' | 'permissions' | 'revoked'
+  | 'hierarchyId'
+  | 'parentId'
+  | 'childrenIds'
+  | 'account'
+  | 'permissions'
+  | 'revoked'
 >
 
 export type DelegationNodeId = Hash
 
 export interface IChainDelegationNode extends Struct {
-  readonly rootId: DelegationNodeId
+  readonly hierarchyRootId: DelegationNodeId
```

Comment:
And also on the mashnet node. I would simply leave it for now, honestly.

Reply:
At the end it is not semantically wrong, as the hierarchy ID === root node ID.

Reply:
true, we could describe it as (hierarchy/root node) id. In fact, technically, you wouldn't have to mess with the mashnet node though, because we can choose field names at will with the types, and they don't have to match anything on the mashnet implementation. The mapping is purely positional, just the type needs to be right.